pkg/controller/jobs/job/job_controller_test.go

```diff
@@ -1954,6 +1955,7 @@ func TestReconciler(t *testing.T) {
 			wantJob: *baseJobWrapper.Clone().
 				Suspend(true).
 				Active(10).
+				PodLabel("testingKey", "testingVal").
```

Comment:
The outcome is expected.
Since the workload does not contain those labels in pods template, the restored job does not contain them too.
The test is designed in a way to create workloads directly not from the job - [line 2747](https://github.com/kubernetes-sigs/kueue/blob/a54f3660182f1a5277c9bfc76c175cf95d5321b1/pkg/controller/jobs/job/job_controller_test.go#L2747)
Thus wantJob can only contain the labels that workload had.

Reply:
Oh, you're right. In this unit testing, we do not perform the workload controller. So, the corresponding workload will be created based on the input data.

So, this is not regression. Thank you for the investigation!

Reply:
/close